ui/api.py

```diff
@@ -53,19 +55,42 @@ def process_transcode_results(video, job):
             )
 
 
-def refresh_status(video):
+def refresh_status(video, encode_job=None):
     """
     Check the encode job status & if not complete, update the status via a query to AWS.
 
     Args:
         video(ui.models.Video): Video object to refresh status of.
+        encode_job(dj_elastictranscoder.models.EncodeJob): EncodeJob associated with Video
     """
-    if video.status not in ['Complete', 'Error']:
+    if video.status in [VideoStatus.TRANSCODING]:
         try:
-            job = get_et_job(video.encode_jobs.latest("created_at").id)
-            if job['Status'] == 'Complete':
-                process_transcode_results(video, job)
-            video.status = job['Status']
-            video.save()
-        except EncodeJob.DoesNotExist:
-            log.error("No encoding job exists for video id %d", video.id)
+            if not encode_job:
+                encode_job = video.encode_jobs.latest("created_at")
+            et_job = get_et_job(encode_job.id)
+            if et_job['Status'] == VideoStatus.COMPLETE:
+                process_transcode_results(video, et_job)
+                encode_job.message = et_job
+                encode_job.save()
+                update_video_status(video, VideoStatus.COMPLETE)
+            elif et_job['Status'] == VideoStatus.ERROR:
+                update_video_status(video, VideoStatus.TRANSCODE_FAILED)
```

Comment:
ClientError could happen in either `get_et_job` or `process_transcode_results` method calls.  I'll make a separate try/catch block for `EncodeJob.DoesNotExist`